core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java

```diff
@@ -67,7 +76,22 @@
     List<DataFileFooter> indexInfo = fileFooterConverter.getIndexInfo(
         identifier.getIndexFilePath() + CarbonCommonConstants.FILE_SEPARATOR + identifier
             .getIndexFileName(), indexFileStore.getFileData(identifier.getIndexFileName()));
+    CarbonTable carbonTable =
+        CarbonMetadata.getInstance().getCarbonTable(identifier.getTableUniqueName());
```

Comment:
Avoid getting carbonTable using CarbonMetadata.getInstance as it can create problem in concurrent scenarios

Reply:
@manishgupta88 : If your concern is that schema can be modified. I can make this carbon table access only for NonTransactional tables as it is required only for NonTransactional table and schema modify (IUD) is not supported for NonTransactional table